### DISCUSSION TITLE
tycho 2.7.2 breaks my build

### DISCUSSION THREAD

**Discussion by StephFo** [eclipse-tycho/tycho#952] [Q&A]
Hi,
Upgrading from Tycho 2.6.0 to 2.7.2, when I compile a bundle that contains many internal jar (some are plain jar, other bundles) used in runtime section, now it fails with this error:
_[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.2:validate-classpath (default-validate-classpath) on project com.toto.dependencies: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:2.7.2:validate-classpath failed: Exception parsing OSGi MANIFEST /m2/repository/org/apache/tomee/javaee-api/8.0-5/javaee-api-8.0-5.jar!/META-INF/MANIFEST.MF: MANIFEST header 'Bundle-Version' not found -> [Help 1]_

Indeed this jar file does not contain the expected Bundle-Version. 

With Tycho 2.6.0 it was not reported as a failure.
Is it possible to switch off this validation please ?

Regards,
Stephane

**Comment by laeubi** [comment]
> Indeed this jar file does not contain the expected Bundle-Version.

Then the message is obviously correct :-)

> With Tycho 2.6.0 it was not reported as a failure.

We constantly improve missing validations

> Is it possible to switch off this validation please ?

Nope.

If you think what you are doing is a valid use-case provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the use-case so we can discuss and integrate it into the tycho-test suite. Everything not covered by the test-suite is mostly undefined behavior and prone to regressions.

**Reply by StephFo** [reply]
Hi,
Thank you for explanation and quick answer.

I did not mention that my bundle, that embeds jars file, has a pom with standard maven dependencies to fetch the embedded jars.

Since the pomDependencies was set to 'consider' in parent pom, that is the reason that explains the issue with tycho 2.7.x

In consequence, I have overridden that in the bundle pom file and it fixes my issue.

Here is the declaration to add :
```
            <plugin>
                <groupId>org.eclipse.tycho</groupId>
                <artifactId>target-platform-configuration</artifactId>
                <version>${tycho.version}</version>
                <configuration>
                    <executionEnvironment>${java-executionEnvironment}</executionEnvironment>
                    <resolver>p2</resolver>
                    <pomDependencies>**ignore**</pomDependencies>
                </configuration>
            </plugin>
```

**Reply by laeubi** [reply]
> Since the pomDependencies was set to 'consider' in parent pom, that is the reason that explains the issue with tycho 2.7.x

Yes this sounds reasonable. Please note that Tycho 2.7.x has improved support for mixed reactor builds now: https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md#mixed-reactor-build-support

**Reply by StephFo** [reply]
Yes I know and that is really great 👍 

But sometimes you need to create by your own such a wrapper bundle because all embedded jars require to use the same shared class loader. 
Then I don't want one bundle per jar but a single big one that contains all non OSGi designed 3rd parties jars.

Stéphane
